### ISSUE TITLE
Button action list: add "Do Nothing" (ie. want to disable Button 4 and Button 4) [enhancement]

### ISSUE THREAD

**Issue by sprive** [noah-nuebling/Mac-Mouse-Fix#568]
## Description
I wish to disable Button 4 and Button 5.  This choice is not available and I do not see any alternate way to do this.

## Rationale
My mouse has poor placement of Button 4 and Button 5, such that in normal browsing I am often thumb-pressing these buttons by accident. In a browser this is rather impactful (page back and forward by default)

I can achieve "Do nothing" by mapping both buttons to middle-click, and this way those button clicks effectively do nothing since _most_ mac apps ignore that.

## Additional info
This workaround is OK... for apps that don't support middle-click. I know that for any other apps (that respond to middle click) I can do a per-application mapping of buttons 4 and 5 to go to a non-existant  shortcut key. 

FYI the mouse is a "steelseries Rival3 wireless" and works surprising well on the mac, aside from the low placement of buttons 4   5.

**Comment by ghost** [comment]
在4键和5键那里点“-”去除按键功能，不就可以了吗？

**Comment by brechtm** [comment]
I currently achieve this effect by assigning a many-key shortcut such as <kbd>^</kbd><kbd>⌥</kbd><kbd>⌘</kbd><kbd>F1</kbd> that is unlikely to do anything.

**Comment by sprive** [comment]
@brechtm   The way you worked around the issue is an improvement over my approach. Thanks for sharing, cheers.

**Comment by brechtm** [comment]
It will still cancel a drag that might be in progress, unfortunately.

**Comment by brechtm** [comment]
I'd also like to be able to disable e.g. Middle Button _Click and Drag_, since that has a native functionality in some applications.

EDIT: I discovered that it's possible to remove the entry for _Click and Drag_, but then Mac Mouse Fix still captures it (it becomes a no-op).

**Comment by sprive** [comment]
Ohhh this exchange really helped.
@Javen-Yen tried to tell me to use "-" to delete a button.item, but when I tried that key (-) it did not remove the item. So I gave up.

 @brechtm You just confirmed you could delete a button, which made me re-examine. I realized you needed to (first)  RIGHT-CLICK the rows for "Button 4", and (second) CLICK on the pop-up that contained a `[-]` 
Then the button4 is deleted.

I hope writing it all out longhand like this helps the next person. This ties into my original request "Do nothing" would save people from needing to delete the button (since it's a more drastic step), but this is fine for me.  TL;DR this could use some UI hinting.

**Comment by brechtm** [comment]
> I realized you needed to (first) RIGHT-CLICK the rows for "Button 4", and (second) CLICK on the pop-up that contained a `[-]`

There are also <kbd>+</kbd> and <kbd>-</kbd> buttons below the list view. <kbd>-</kbd> will delete the currently selected item.

Deleting the Button 4/5 _Click_ items simply restores macOS's default action for these (back and forward navigation).
Deleting the Middle Button _Click and Drag_ item, however, doesn't! It breaks middle-mouse button click-and-drag that works fine when Mac Mouse Fix is disabled.

**Comment by sprive** [comment]
So I spoke too soon - "deleting" buttons 4 and 5 simply took those buttons out of control of Mac Mouse Fix.
That reverts mouse behavior to what I had when I went looking for an application like Mac Mouse Fix: my fat thumb kept pressing button4 and button5. In many applications you do not want to randomly push these buttons...

And the best workaround I have is my original mapping of buttons 4 and 5, to "Middle Click". Or better, @brechtm's suggestion to map to  less-often used `^⌥⌘F1`

To summarize the request: there is need for a UI selection that lets you "blackhole" all of a button's activity. (Or if the OS does not allow that, then offer to map the button to an "improbably used" keyboard mapping, such as the one Brecht suggested).  Cheers.

**Comment by brechtm** [comment]
I don't know why I didn't think of this earlier, but Hammerspoon can do this, of course. It's actually better, since it can disable buttons completely; "complex-shortcut disables" don't interfere with an in-progress drag and middle mouse drag still works. See my [StackOverflow answer](https://superuser.com/a/1788257/71891).

Mac Mouse Fix is of course much more user friendly, but since I'm using Hammerspoon for several other purposes, this saves running yet another tool.
